### PR TITLE
feat(fishing-license): oidc token exchange

### PR DESCRIPTION
**Pull request by bensijons**
# Configure token exchange

Attach a link to issue if relevant

## What

Setup token exchange for OIDC

## Why

For authenticating with the Fiskistofa API

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/clients/fishing-licence/src/lib/FishingLicenseApiFactoryConfig.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+export const FishingLicenseApiFactoryConfig = (
+  xRoadConfig: ConfigType<typeof XRoadConfig>,
+  config: ConfigType<typeof FishingLicenseClientConfig>,
+  idsClientConfig: ConfigType<typeof IdsClientConfig>,
+) =>
+  new Configuration({
+    fetchApi: createEnhancedFetch({
+      name: 'clients-fishing-license',
+      treat400ResponsesAsErrors: true,
```
Comment (eirikurn) [inline libs/clients/fishing-licence/src/lib/FishingLicenseApiFactoryConfig.ts:18]: Why?

File: libs/clients/fishing-licence/src/lib/FishingLicenseApiFactoryConfig.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      logErrorResponseBody: true,
+      autoAuth: idsClientConfig.clientSecret
```
Comment (eirikurn) [inline libs/clients/fishing-licence/src/lib/FishingLicenseApiFactoryConfig.ts:20]: Better to check `idsClientConfig.isConfigured`.

File: libs/clients/fishing-licence/src/lib/SkipApiProvider.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+export const SkipApiProvider: Provider<SkipApi> = {
+  provide: SkipApi,
+  scope: LazyDuringDevScope,
+  useFactory: (
+    xRoadConfig: ConfigType<typeof XRoadConfig>,
+    config: ConfigType<typeof FishingLicenseClientConfig>,
+    idsConfig: ConfigType<typeof IdsClientConfig>,
+  ) =>
+    new SkipApi(FishingLicenseApiFactoryConfig(xRoadConfig, config, idsConfig)),
```
Comment (eirikurn) [inline libs/clients/fishing-licence/src/lib/SkipApiProvider.ts:20]: I guess you saw the other way to register multiple APIs in the NationalRegistryV2 client.

**Review by eirikurn — APPROVED** [review]
Sweet!

File: libs/clients/fishing-licence/src/lib/SkipApiProvider.ts (hunk 3 above)
Comment (bensijons) [reply libs/clients/fishing-licence/src/lib/SkipApiProvider.ts:20]: I did not, but now that I have taken a look I wish I had thought of that 😄 I'll change into doing it same way as NationalRegistryV2, feels cleaner

File: libs/clients/fishing-licence/src/lib/FishingLicenseApiFactoryConfig.ts (hunk 1 above)
Comment (bensijons) [reply libs/clients/fishing-licence/src/lib/FishingLicenseApiFactoryConfig.ts:18]: Good question. Unnecessary here I believe, removed

**Pull request merged**